pkg/apis/crd/v1alpha1/types.go

```diff
@@ -709,3 +709,145 @@ type ExternalNodeList struct {
 
 	Items []ExternalNode `json:"items,omitempty"`
 }
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type SupportBundleCollectionList struct {
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ListMeta `json:"metadata,omitempty"`
+
+	Items []SupportBundleCollection `json:"items"`
+}
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type SupportBundleCollection struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard metadata of the object.
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Specification of the desired behavior of SupportBundleCollection.
+	Spec SupportBundleCollectionSpec `json:"spec"`
+	// Most recently observed status of the SupportBundleCollection.
+	Status SupportBundleCollectionStatus `json:"status"`
+}
+
+type SupportBundleCollectionSpec struct {
+	Nodes         *BundleNodes         `json:"nodes,omitempty"`
+	ExternalNodes *BundleExternalNodes `json:"externalNodes,omitempty"`
+	// ExpirationMinutes is the requested duration of validity of the SupportBundleCollection.
+	// A SupportBundleCollection will be marked as Failed if it does not finish before expiration.
+	// Default is 60.
+	ExpirationMinutes int32 `json:"expirationMinutes"`
+	// SinceTime specifies a relative time before the current time from which to collect logs
+	// A valid value is like: 1d, 2h, 30m.
+	SinceTime      string                        `json:"sinceTime,omitempty"`
+	FileServer     BundleFileServer              `json:"fileServer"`
+	Authentication BundleServerAuthConfiguration `json:"authentication"`
+}
+
+// BundlePhase defines the phase in which a SupportBundleCollection is.
+type BundlePhase string
+
+const (
+	// BundlePending means the SupportBundleCollection has been accepted by the system,
+	// but it has not been processed by Antrea.
+	BundlePending BundlePhase = "Pending"
+	// BundleProcessing means the SupportBundleCollection is being processed by Antrea.
+	BundleProcessing BundlePhase = "Processing"
+	// BundleSucceeded means the SupportBundleCollection is successfully processed by all
+	// required Nodes or ExternalNodes.
+	BundleSucceeded BundlePhase = "Succeeded"
+	// BundlePartiallySucceeded means some required Nodes or ExternalNodes have successfully
+	// processed the SupportBundleCollection request, but others failed.
+	BundlePartiallySucceeded BundlePhase = "PartiallySucceeded"
+	// BundleFailed means the SupportBundleCollection failed. It can happen when Antrea
+	// Controller fails to create internal objects, or all of the required Nodes or
+	// ExternalNodes fail to process the request.
+	BundleFailed BundlePhase = "Failed"
+)
+
+type SupportBundleCollectionStatus struct {
+	Phase BundlePhase `json:"phase"`
```

Comment:
It's suggested to use conditions instead of phase for new APIs: https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#typical-status-properties
> Some resources in the v1 API contain fields called phase, and associated message, reason, and other status fields. The pattern of using phase is deprecated. Newer API types should use conditions instead. Phase was essentially a state-machine enumeration field, that contradicted [system-design principles](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/architecture/principles.md#control-logic) and hampered evolution, since [adding new enum values breaks backward compatibility](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api_changes.md). Rather than encouraging clients to infer implicit properties from phases, we prefer to explicitly expose the individual conditions that clients need to monitor. Conditions also have the benefit that it is possible to create some conditions with uniform meaning across all resource types, while still exposing others that are unique to specific resource types. See [#7856](http://issues.k8s.io/7856) for more details and discussion.

And the conditions you have added can already represent some phases, e.g. "started=false" means "Pending", "started=true" means "Processing", maybe you could add one condtion "Succeeded" to represent the result.

Reply:
Since we also have a special result "partiallySucceeded",  I don't think "Succeeded" condition can present these three results: Succeeded, PartiallySucceeded, Failed. Any good suggestion?

Reply:
I think Deployment's "ReplicaFailure" is similar to it: https://github.com/kubernetes/kubernetes/blob/a9593d634c6a053848413e600dadbf974627515f/pkg/apis/apps/types.go#L532-L534
```
// ReplicaFailure is added in a deployment when one of its pods fails to be created
// or deleted.
DeploymentReplicaFailure DeploymentConditionType = "ReplicaFailure"
```
It could use something like "CollectionFailure" which would be true when one of the collections fails?

I also feel users care more about whether the collection is finished rather than whether it is started, so perhaps a condition like "Complete" is more useful than "Started", similar to Job's conditions: https://github.com/kubernetes/kubernetes/blob/a9593d634c6a053848413e600dadbf974627515f/pkg/apis/batch/types.go#L312-L313
```
// JobComplete means the job has completed its execution.
JobComplete JobConditionType = "Complete"
```
@jianjuns do you think we should follow the API convention to add only Conditions (no phase) for new APIs?

Reply:
I thought about using conditions only too, but as I knew K8s resources like Pod have phases too, I did not suggest to make a change. But if the K8s new convention is to use conditions only, then I would agree we should follow that.

I thought about how to express "partial success" with conditions too. Maybe we can have the following conditions:
- "Completed"
- "CollectionFailure" - when there is a failure.
- "BundleCollected" - when at least one Node upload the bundle.

Reply:
I don't think "Started" is conflict with "Completed", and we can have both. I want to use "Started" as a condition to express if Antrea Controller has successfully created the internal object, and we can use this condtion if any error happens before Controller syncs the internal objects to Agents. For "Completed", it is a final condtion which is set if all required Nodes/ExternalNodes have successfully upload the files. Like Pod condtions, both PodReady and ContainerReady are existing. @tnqn @jianjuns

Reply:
@wenyingd could "CollectionFailure" be used to show that error? I feel creating the internal object is an internal step, having a separate condition to hold its error makes the status a bit hard to consume, consumers need to check one condition for some types of collection failure and check another condition for other types of collection failures.

Reply:
I feel "internal object created" is not for regular users. If we add "Started", it should be for giving users a feedback that "Antrea has started to handle the collection request". I am not very against to the idea.

And in my mind ”Completed" should not mean "all Nodes have successfully uploaded the bundle" (or no Node is selected by the CR), but means "Antrea has finished processing the collection". If we want to express all Nodes succeeded to upload the bundle, we should add a condition like "Succeeded", but that seems the same thing with "CollectionFailure = false and Completed = true".

Reply:
I still feel we need a condition "Started" to identify if the request is started. It is not only for end users, but also needed by Antrea. Let's think about this scenario, Antrea Controller is processing one support bundle request, but it is not completed, then Controller is restarted. In theory, Controller should continue the original request. But if we do not have such a flag or condtion to mark it, Controller can not know this, and it is possible to start from a different request. 

Besides, Antrea Controller will only processes one bundle collection request in the same namespace at one time, this is helpful to avoid a duplication on Agent processing bundle request from two CR resources. This means the processing time on a supportbunlde request is long. With a condition Started=false and reason is "another support bundle collection in Namespace XX is processing" is kind to users to know why the request is still not completed.